src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int FIGURE_COUNT = 5;
+    private static final int DEFAULT_VALUE = 10;
+    private static final Color COLOR_WHITE = Color.WHITE;
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int figureNumber = random.nextInt(FIGURE_COUNT);
+        String color = colorSupplier.getRandomColor();
+
+        int radius = random.nextInt(FIGURE_COUNT);
+        int sideOne = random.nextInt(FIGURE_COUNT);
+        int sideTwo = random.nextInt(FIGURE_COUNT);
+        int length = random.nextInt(FIGURE_COUNT);
```

Comment:
The variables `radius`, `sideOne`, `sideTwo`, and `length` are initialized regardless of the figure to be created. This might lead to unnecessary computations if, for example, a `Circle` is to be created (which requires only `radius`). Consider initializing these variables inside the corresponding `case` blocks.